### ISSUE TITLE
Eslint Setup Help

### ISSUE THREAD

**Issue by KasonBraley** [mfussenegger/nvim-lint#69]
I am having trouble setting this plugin up so I can lint JS files with `eslint`. Would you be able to provide so examples or any pointers on how to get started? I don't see anywhere on the Readme about how the errors from the linter are displayed, is that handled by neovim's LSP instead or something else I would need to setup?

Here is my `nvim-lint` config. 
```
local lint = require "lint"

lint.linters_by_ft = {
  javascript = {"eslint"}
}

vim.cmd([[au BufEnter *.js lua require('lint').try_lint() ]])
vim.cmd([[au BufWritePost *.js lua require('lint').try_lint() ]])
```

Some things I have tried to get it to work:

- I have tried using eslint as both a global package as well as a dev dependency
- Explicitly specified elsint's args like mentioned in #31
- tried setting `lint.linters.eslint.cmd = "node_modules/.bin/eslint"` when i installed the eslint package as a dev dependency

Any help is appreciated!

**Comment by mfussenegger** [comment]
Yes the diagnostics are reported via neovim's LSP diagnostic module. By default you should get signs displayed, highlights and in-line messages via virtual-text. You can customize that. See `:help vim.lsp.diagnostic.on_publish_diagnostics()`.

Your config looks okay.

> tried setting lint.linters.eslint.cmd = "node_modules/.bin/eslint" when i installed the eslint package as a dev dependency

I think the path needs to be absolute. 

You can try running `:lua require('lint').try_lint('eslint')` manually to see if it works or if there is an error.

**Comment by mfussenegger** [comment]
> I did that and I received `Linter command ./node_modules/.bin/eslint exited with code 1`

Looks like the linter definition would need the `ignore_exitcode = true,` option.
Could you post some sample linter output in text form? I don't use eslint myself but that would allow me to create a test case.

**Comment by heygarrett** [comment]
I'm also not seeing diagnostics from eslint.

Current config:  
```lua
local lint = require('lint')
lint.linters.eslint.cmd = './node_modules/.bin/eslint'
lint.linters_by_ft = {
	javascript = {'eslint'},
	typescript = {'eslint'}
}
vim.cmd([[au BufWritePost <buffer> lua require('lint').try_lint()]])
```

<img width="305" alt="Screen Shot 2021-09-01 at 18 37 29" src="https://user-images.githubusercontent.com/1154912/131754432-bb5f5397-50eb-4a7c-a4d1-8e616c16db53.png">

I also tried running `:lua require('lint').try_lint('eslint')` and got `Linter command ./node_modules/.bin/eslint exited with code 1` (and same thing for global eslint, if I don't manually set the command).

In my terminal:  
```
❯ node_modules/.bin/eslint test.ts

/Users/garrett/repos/TypeScript/test_project/test.ts
  1:5  error  'message' is never reassigned. Use 'const' instead  prefer-const

✖ 1 problem (1 error, 0 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.
```

I'm new to TypeScript and JavaScript, but I'm happy to provide any additional information that might be helpful.

**Comment by heygarrett** [comment]
I just realized I get an additional error message from `:messages`:  

    Error executing vim.schedule lua callback: ...w/Cellar/neovim/0.5.0/share/nvim/runtime/lua/vim/uri.lua
    :116: attempt to index local 'uri' (a nil value)
    Linter command ./node_modules/.bin/eslint exited with code 1

**Comment by heygarrett** [comment]
This closed issue in the Vala language server repo looks similar:  
https://github.com/Prince781/vala-language-server/issues/169

**Comment by mfussenegger** [comment]
> I just realized I get an additional error message from `:messages`:
> 
> ```
> Error executing vim.schedule lua callback: ...w/Cellar/neovim/0.5.0/share/nvim/runtime/lua/vim/uri.lua
> :116: attempt to index local 'uri' (a nil value)
> Linter command ./node_modules/.bin/eslint exited with code 1
> ```

This is a new unrelated error and has to do with https://github.com/mfussenegger/nvim-lint/pull/82
If you're using neovim from master, you'll have to update it.

**Comment by heygarrett** [comment]
> If you're using neovim from master, you'll have to update it.

I'm using the stable from Homebrew:  
> ❯ nvim --version  
> NVIM v0.5.0  
> Build type: Release

**Comment by mfussenegger** [comment]
> > If you're using neovim from master, you'll have to update it.
> 
> I'm using the stable from Homebrew:
> 
> > ❯ nvim --version
> > NVIM v0.5.0
> > Build type: Release

Should work again with https://github.com/mfussenegger/nvim-lint/pull/91

**Comment by heygarrett** [comment]
@KasonBraley Does my pull request resolve the issue for you? I was trying to fix what I thought was an unrelated issue, but I think both had the same root cause.

**Comment by KasonBraley** [comment]
@heygarrett yes I've got eslint diagnostics showing correctly, thank you.
